doc/nrf/drivers/mspi_sqspi.rst

```diff
@@ -0,0 +1,192 @@
+				ranges = <0 0x2003c000 0x4000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				sqspi: sqspi@3c00 {
+					compatible = "nordic,nrf-sqspi";
+					#address-cells = <1>;
+					#size-cells = <0>;
+					reg = <0x3c00 0x200>;
+					status = "okay";
+					zephyr,pm-device-runtime-auto;
+				};
+			};
+		};
+	};
+
+For the nRF54H20 SoC, additionally a region for data buffers to be used in
+sQSPI transfers must be reserved in RAM accessible by the FLPR core. This
+region needs to be indicated to the shim driver by the ``memory-regions``
+property in the node representing the sQSPI. If an MSPI transfer will be
+requested with a buffer located outside of this region, the shim driver will
+allocate a buffer in the region for the time of the transfer and do the proper
+copying of data between these two buffers (before a TX transfer or after an RX
+transfer). To avoid the overhead of this automatic allocation and copying,
+a proper buffer can be allocated statically using :c:macro:`DMM_MEMORY_SECTION`.
+Then the shim driver will pass such buffer directly to the sQSPI.
+
+An example configuration for the nRF54H20 SoC:
+
+.. code-block:: devicetree
+
+	&pinctrl {
+		sqspi_default: sqspi_default {
+			group1 {
+				psels = <NRF_PSEL(SDP_MSPI_SCK, 7, 0)>,
+					<NRF_PSEL(SDP_MSPI_CS0, 7, 5)>,
+					<NRF_PSEL(SDP_MSPI_DQ0, 7, 1)>;
+				nordic,drive-mode = <NRF_DRIVE_E0E1>;
+			};
+			group2 {
+				psels = <NRF_PSEL(SDP_MSPI_DQ1, 7, 2)>,
+					<NRF_PSEL(SDP_MSPI_DQ2, 7, 3)>,
+					<NRF_PSEL(SDP_MSPI_DQ3, 7, 4)>;
+				nordic,drive-mode = <NRF_DRIVE_E0E1>;
+				bias-pull-up;
+			};
+		};
+
+		sqspi_sleep: sqspi_sleep {
+			group1 {
+				low-power-enable;
+				psels = <NRF_PSEL(SDP_MSPI_SCK, 7, 0)>,
+					<NRF_PSEL(SDP_MSPI_CS0, 7, 5)>,
+					<NRF_PSEL(SDP_MSPI_DQ0, 7, 1)>,
+					<NRF_PSEL(SDP_MSPI_DQ1, 7, 2)>,
+					<NRF_PSEL(SDP_MSPI_DQ2, 7, 3)>,
+					<NRF_PSEL(SDP_MSPI_DQ3, 7, 4)>;
+			};
+		};
+	};
+
+	&cpuflpr_vpr {
+		pinctrl-0 = <&sqspi_default>;
+		pinctrl-1 = <&sqspi_sleep>;
+		pinctrl-names = "default", "sleep";
+		interrupts = <212 NRF_DEFAULT_IRQ_PRIORITY>;
+		status = "okay";
+		/delete-property/ execution-memory;
+	};
+
+	/delete-node/ &cpuflpr_code_data;
+	/delete-node/ &cpuapp_cpuflpr_ipc_shm;
+	/delete-node/ &cpuflpr_cpuapp_ipc_shm;
+	/delete-node/ &cpuapp_cpuflpr_ipc;
+
+	&ram21_region {
+		status = "okay";
+
+		softperiph_ram: memory@0 {
+			reg = <0 0x4000>;
+			ranges;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			sqspi: sqspi@3e00 {
+				compatible = "nordic,nrf-sqspi";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0x3e00 0x200>;
+				zephyr,pm-device-runtime-auto;
+				memory-regions = <&sqspi_buffers>;
+			};
+		};
+
+		sqspi_buffers: memory@4000 {
+			compatible = "zephyr,memory-region";
+			reg = <0x4000 0x4000>;
+			#memory-region-cells = <0>;
+			zephyr,memory-region = "SQSPI_BUFFERS";
+			zephyr,memory-attr = <DT_MEM_CACHEABLE>;
+		};
+	};
+
+API Reference
+*************
+
```

Comment:
Is there any header/source file path that can be added here? (example: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/drivers/sensor_stub.html#api_documentation )

Reply:
Added a source file.